lib/modules/datasource/github-releases/cache/cache-base.ts

```diff
@@ -244,66 +263,56 @@ export abstract class AbstractGithubDatasourceCache<
           : this.maxPrefetchPages;
         let stopIteration = false;
         while (pagesRemained > 0 && !stopIteration) {
-          const graphqlRes = await this.http.postJson<
-            GithubGraphqlResponse<QueryResponse<FetchedItem>>
-          >('/graphql', {
-            baseUrl,
-            body: { query: this.graphqlQuery, variables },
-          });
+          const res = await this.query(baseUrl, variables);
+          if (res instanceof Error) {
+            throw res;
```

Comment:
Next step: check error message starts with specific error message and shrink page size for the next loop iteration